iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/thread.inl

```diff
@@ -0,0 +1,40 @@
+#ifndef IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+#define IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+
+#include "iceoryx_hoofs/posix_wrapper/thread.hpp"
+
+namespace iox
+{
+namespace posix
+{
+inline cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread,
+                                                        const cxx::function<void()>& callable) noexcept
+{
+    if (!callable)
+    {
+        std::cerr << "The thread cannot be created with an empty callable." << std::endl;
+        return cxx::error<ThreadError>(ThreadError::EMPTY_CALLABLE);
+    }
+
+    uninitializedThread.emplace();
+    uninitializedThread->m_callable = callable;
+
+    // create thread
+    auto createResult =
+        posixCall(pthread_create)(
+            &uninitializedThread->m_threadHandle, nullptr, Thread::startRoutine, &uninitializedThread->m_callable)
```

Comment:
Could you create a `const pthread_attr_t * threadAttributes = nullptr` and use it here instead of a `nullptr`.
In a follow up PR you can use this variable and set all the thread attributes but for now it would make the core more readble.